app/_features/event/components/event-form.tsx

```diff
@@ -0,0 +1,493 @@
+      }
+    },
+    []
+  );
+
+  useEffect(() => {
+    if (!user) document.dispatchEvent(new CustomEvent('open:sign-in-modal'));
+  }, [user]);
+
+  useEffect(() => {
+    if (eventName.trim().length === 0) {
+      setTitleValid(false);
+    } else {
+      setTitleValid(true);
+    }
+  }, [eventName]);
+
+  useEffect(() => {
+    if (eventDescription.trim().length === 0) {
+      setDescValid(false);
+    } else {
+      setDescValid(true);
+    }
+  }, [eventDescription]);
+
+  useEffect(() => {
+    if (startTime.hour > endTime.hour) {
+      setEndTime({
+        hour: `${startTime.hour}`,
+        minute: `${startTime.minute}`,
+      });
+    }
+  }, [startTime.hour, startTime.minute, endTime.hour]);
+
+  const createEvent = useCallback(
+    async (isDraft: boolean) => {
+      if (
+        eventDescription.trim().length === 0 ||
+        eventName.trim().length === 0
+      ) {
+        document.dispatchEvent(new CustomEvent(incompleteFieldEvent));
+        return;
+      }
+
+      const finalEndpoint = `/api/events/create`;
+
+      const eventStartTime = new Date(
+        date.setHours(parseInt(startTime.hour), parseInt(startTime.minute), 0)
+      );
+      const eventEndTime = new Date(
+        date.setHours(parseInt(endTime.hour), parseInt(endTime.minute), 0)
+      );
+
+      let imageFile: File | Blob | null = null;
+
+      if (imageData.imageBlob) {
+        imageFile = await compressImage(imageData.imageBlob, 280, 560, 0.8);
+      }
+
+      const data = JSON.stringify({
+        name: eventName,
+        startTime: eventStartTime.toISOString(),
+        endTime: eventEndTime.toISOString(),
+        description: eventDescription.replace(/(?:\r\n|\r|\n)/g, '<br>'),
+        host: user?.name,
+        isPublished: !isDraft,
+      });
+
+      const formData = new FormData();
+
+      if (imageFile) {
+        formData.append('image', imageFile, 'poster.webp');
+      }
+      formData.append('data', data);
+
+      const respEvent = await InternalApiFetcher.post(finalEndpoint, {
+        body: formData,
+        headers: undefined,
+      });
+
+      if (respEvent.ok) {
+        const redirectPath = new URL(
+          `/event/${respEvent.data.slug}`,
+          window.location.origin
+        ).href;
+        navigateTo(redirectPath);
+      } else {
+        console.log(respEvent.message);
+      }
+    },
+    [
+      date,
+      endTime.hour,
+      endTime.minute,
+      eventDescription,
+      eventName,
+      imageData,
+      navigateTo,
+      startTime.hour,
+      startTime.minute,
+      user?.name,
+    ]
+  );
+
+  const saveAsDraft = useCallback(() => {
+    createEvent(true);
+  }, [createEvent]);
+
+  const updateDraft = useCallback(() => {
+    console.log('update draft');
+  }, []);
+
+  const onPublish = useCallback(() => createEvent(true), [createEvent]);
+
+  const updatePublishedEvent = useCallback(() => {
+    console.log('update published event');
+  }, []);
```

Comment:
If you intend to  continue `updateDraft` and `updatePublishedEvent` later, I think better to change this with descriptive comment.

Reply:
updated with comments instead : https://github.com/inlivedev/inlive-room/pull/130/commits/ecb61927672c4fa4bab051eaba5a36f3f471bc4d